FirstClasses/average/src/average.erl

```diff
@@ -3,4 +3,49 @@
 -export([average/1]).
 
 average(_List) ->
-    ok.
+    case _List of 
+        [] ->
+            0;
+        _ ->
+            sum(_List) / list_lenght(_List)
+    end.
+
+sum(List) ->
+    %sum_buildin(List).
+    sum_recursion(List).
+
+list_lenght(List) ->
+    %list_lenght_buildin(List).
+    list_lenght_recursion(List).
+
+sum_buildin(List) ->
+    lists:sum(List).
+
+sum_recursion([H|T]) ->
```

Comment:
How about matching a function head like this:
```erlang
sum_recursion([]) -> 0;
sum_recursion([H | T]) ->
    H + sum_recursion(T).

```

Reply:
after few more hours of practice with Erlang I agree, this would be more elegant solution